pkg/services/mssqlfg/plan_details.go

```diff
@@ -0,0 +1,285 @@
+	// Basic tier is constrained to just 5 DTUs, if this is the basic tier, there
+	// is no dtus param. We can infer this is the case if the tier details don't
+	// tell us there's a choice.
+	if len(d.allowedDTUs) == 0 {
+		return d.skuMap[d.defaultDTUs]
+	}
+	return d.skuMap[pp.GetInt64("dtus")]
+}
+
+type vCorePlanDetails struct {
+	tierName      string
+	tierShortName string
+}
+
+func (v vCorePlanDetails) validateUpdateParameters(
+	instance service.Instance,
+) error {
+	return validateStorageUpdate(
+		*instance.ProvisioningParameters,
+		*instance.UpdatingParameters,
+	)
+}
+
+func (v vCorePlanDetails) getUpdateSchema() service.InputParametersSchema {
+	ips := service.InputParametersSchema{
+		PropertySchemas: map[string]service.PropertySchema{},
+	}
+	ips.PropertySchemas["cores"] = &service.IntPropertySchema{
+		AllowedValues: []int64{2, 4, 8, 16, 24, 32, 48, 80},
+		DefaultValue:  ptr.ToInt64(2),
+		Description:   "A virtual core represents the logical CPU",
+	}
+	ips.PropertySchemas["storage"] = &service.IntPropertySchema{
+		MinValue:     ptr.ToInt64(5),
+		MaxValue:     ptr.ToInt64(1024),
+		DefaultValue: ptr.ToInt64(10),
+		Description:  "The maximum data storage capacity (in GB)",
+	}
+	return ips
+}
+
+func (v vCorePlanDetails) getProvisionSchema() service.InputParametersSchema {
+	ips := service.InputParametersSchema{
+		RequiredProperties: []string{
+			"failoverGroup",
+			"database",
+		},
+		PropertySchemas: map[string]service.PropertySchema{
+			"failoverGroup": &service.StringPropertySchema{
+				Description: "The name of the failover group",
+			},
+			"database": &service.StringPropertySchema{
+				Description: "The name of the database",
+			},
+			"cores": &service.IntPropertySchema{
+				AllowedValues: []int64{2, 4, 8, 16, 24, 32, 48, 80},
+				DefaultValue:  ptr.ToInt64(2),
+				Description:   "A virtual core represents the logical CPU",
+			},
+			"storage": &service.IntPropertySchema{
+				MinValue:     ptr.ToInt64(5),
+				MaxValue:     ptr.ToInt64(1024),
+				DefaultValue: ptr.ToInt64(10),
+				Description:  "The maximum data storage capacity (in GB)",
+			},
+		},
+	}
+	return ips
+}
+
+func (v vCorePlanDetails) getTierProvisionParameters(
+	pp service.ProvisioningParameters,
+) (map[string]interface{}, error) {
+	p := map[string]interface{}{}
+	p["sku"] = v.getSKU(pp)
+	p["tier"] = v.tierName
+	// ARM template needs bytes
+	p["maxSizeBytes"] = pp.GetInt64("storage") * 1024 * 1024 * 1024
+	return p, nil
+}
+
+func (v vCorePlanDetails) getSKU(pp service.ProvisioningParameters) string {
+	return fmt.Sprintf(
+		"%s_Gen5_%d",
+		v.tierShortName,
+		pp.GetInt64("cores"),
+	)
+}
+
+func getDBMSPairRegisteredUpdateParamSchema() service.InputParametersSchema {
+	return service.InputParametersSchema{
+		SecureProperties: []string{
+			"primaryAdministratorLoginPassword",
+			"secondaryAdministratorLoginPassword",
+		},
+		PropertySchemas: map[string]service.PropertySchema{
```

Comment:
These four fields should be `RequiredProperties`

Reply:
There are updating parameters. Does it make sense to update only one of them (e.g. the primary password)?

Reply:
It makes sense, let's keep it.